spark/so/chain/watch_chain.go

```diff
@@ -719,6 +719,8 @@ func handleBlock(
 				if err != nil {
```

Comment:
good catch. the whole `dbTx.TreeNode...Save(ctx)` returns the updated tree node so you could also update the `treeNode` variable there accordingly

```suggestion
                treeNode, err = dbTx.TreeNode.UpdateOne(treeNode).
					SetStatus(st.TreeNodeStatusAvailable).
					Save(ctx)
```